runtime/common/src/lib.rs

```diff
@@ -417,6 +422,46 @@ where
 	}
 }
 
+/// XcmMessageHandler of `DmpMessageHandler` and `XcmpMessageHandler` implementations.
+/// if xcm is paused, the `max_weight` of each handle method is set to `0`.
+///
+/// Parameters type:
+/// - `A`: `DmpMessageHandler` or `XcmpMessageHandler`
+pub struct XcmMessageHandler<Runtime, A>(PhantomData<(Runtime, A)>);
```

Comment:
This can be moved to transaction pause module. And `T` naming will be more conventional than `A` for generic param.